### PR TITLE
Choose your own adventure medication filtering

### PR DESCRIPTION
**Pull request by captDaylight**
Choose whichever med filter you wanna start with.

This is more of a short term fix, a lot of the current frustration clinicians have with our med search element and catalog lookup.

https://www.notion.so/photons/Rebuild-Advanced-search-Element-short-term-d82c08b2ef4e4569a2327ad1889186d5?pvs=4

**update, some bugs fixed**
- [x] updating concept id doesn't clear the filters (elements)
- [x] product list doesn't clear when you reopen the modal (elements)
- [x] debounce search (elements)

### CODE REVIEW

File: apps/app/src/views/components/AdvancedDrugSearch.tsx [hunk 1]
```diff
@@ -11,265 +11,148 @@ import {
...
-        ref={selectInputRef}
+        name="medicationName"
         options={options}
-        onChange={(e: any) => {
-          setDrugbankPCID(e);
+        setFilterText={setFilterText}
+        filterOption={() => true}
+        placeholder="Search by name"
+        onChange={(id: string) => {
+          setMedId(undefined);
```
Comment (jknapp25) [inline apps/app/src/views/components/AdvancedDrugSearch.tsx:110]: why is this needed?

File: apps/app/src/views/components/AdvancedDrugSearch.tsx [hunk 2]
```diff
@@ -11,265 +11,148 @@ import {
...
+    if (filterText.length > 0) {
+      setMedId(undefined);
+      searchConcepts({ variables: { name: filterText } });
     }
-  }, [drugbankPCID]);
+  }, [filterText]);
 
   useEffect(() => {
-    if (drugbankPCID) {
+    if (data && data.medicationConcepts) {
```
Comment (jknapp25) [inline apps/app/src/views/components/AdvancedDrugSearch.tsx:87]: okay to do `data?.medicationConcepts` ?

**Review by jknapp25 — APPROVED** [review]

File: apps/app/src/views/components/AdvancedDrugSearch.tsx (hunk 2 above)
Comment (captDaylight) [reply apps/app/src/views/components/AdvancedDrugSearch.tsx:87]: 100%

File: apps/app/src/views/components/AdvancedDrugSearch.tsx (hunk 1 above)
Comment (captDaylight) [reply apps/app/src/views/components/AdvancedDrugSearch.tsx:110]: it's not!
